pkg/agent/nodeportlocal/portcache/port_table.go

```diff
@@ -23,26 +23,54 @@ import (
 	"sync"
 	"time"
 
-	"k8s.io/klog/v2"
-
 	"antrea.io/antrea/pkg/agent/nodeportlocal/rules"
+
+	"k8s.io/klog/v2"
 )
 
-type NodePortData struct {
-	NodePort int
-	PodPort  int
-	PodIP    string
+type ProtocolSocketData struct {
+	Protocol string
 	socket   Closeable
 }
 
+type NodePortData struct {
+	NodePort  int
+	PodPort   int
+	PodIP     string
+	Protocols []ProtocolSocketData
+}
+
+func (d *NodePortData) HasProtocol(protocol string) bool {
+	for _, p := range d.Protocols {
+		if p.Protocol == protocol {
+			return true
+		}
+	}
+	return false
+}
+
+func (d *NodePortData) DeleteProtocol(protocol string) error {
+	for i, p := range d.Protocols {
+		if p.Protocol == protocol {
+			d.Protocols = append(d.Protocols[:i], d.Protocols[i+1:]...)
+			if err := p.socket.Close(); err != nil {
+				return fmt.Errorf("Error when releasing local port %d: %v", d.NodePort, err)
```

Comment:
```suggestion
				return fmt.Errorf("error when releasing local port %d: %v", d.NodePort, err)
```

we don't capitalize these